Mathlib/Combinatorics/SimpleGraph/Connectivity.lean

```diff
@@ -1016,8 +1015,6 @@ theorem isCycle_copy {u u'} (p : G.Walk u u) (hu : u = u') :
   rfl
 #align simple_graph.walk.is_cycle_copy SimpleGraph.Walk.isCycle_copy
 
-lemma IsCycle.not_Nil {p : G.Walk v v} (hp : IsCycle p) : ¬ p.Nil := (hp.ne_nil ·.eq_nil)
```

Comment:
You shouldn't drop this lemma. It's generally useful

Reply:
Oops, that was an accident

Reply:
While you're at it, it should be called `IsCycle.not_nil`, not `IsCycle.not_Nil`.